### PR TITLE
Adding a new test case PartialBackupWithLowerStorkVersion

### PR DESCRIPTION
**Pull request by mkoppal-px**
**What this PR does / why we need it**:

- [x] Added a new test case `PartialBackupWithLowerStorkVersion`
- [x] Created a new helper function to downgrade the stork version - `DowngradeStorkVersion`

**Which issue(s) this PR fixes** (optional)
Closes #PB-7324

**Special notes for your reviewer**:
[Jenkins BYOC on OCP](https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/5969/)

### CODE REVIEW

**Review by sn-px — APPROVED** [review]
LGTM

**Review by dbhatnagar-px — APPROVED** [review]
Lgtm

File: tests/backup/backup_partial_test.go [hunk 1]
```diff
@@ -991,3 +992,237 @@ var _ = Describe("{BackupStateTransitionForScheduledBackups}", Label(TestCaseLab
...
+		}()
+		defer func() {
+			log.InfoD("switching to default context")
+			err := SetClusterContext("")
+			log.FailOnError(err, "failed to SetClusterContext to default cluster")
+		}()
+		err := SetDestinationKubeConfig()
+		log.FailOnError(err, "Switching context to destination cluster failed")
+		log.Infof("Upgrading stork version on destination cluster to %s ", CurrentStorkImage)
+		err = UpgradeStorkVersion(CurrentStorkImage)
```
Comment (kshithijiyer-px) [inline tests/backup/backup_partial_test.go:1205]: Can we put this in a defer? As We would want to make sure that the stork image is updated even when the testcase fails?

Comment (mkoppal-px) [reply tests/backup/backup_partial_test.go:1205]: Even if the test case fails, upgrading stork will be the first thing that will be executed. This is why I have added it as a first step.

Comment (kshithijiyer-px) [reply tests/backup/backup_partial_test.go:1205]: Okay sounds good.

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Pull request merged**